### ISSUE TITLE
Enhance VCF to include "Predicted impact"

### ISSUE THREAD

**Issue by eweitz** [DCGenomics/NCBI_August_Hackathon_Push_Button_Genomics_Solution#13]
Implement @zmartine's proposal to enhance SnpEff's default impact values for each variant with a new "predicted impact" that accounts for significant changes in hydrophobicity, etc.

Input:
- _VCF file annotated by SnpEff_.  This file will be output by the code pushed for [Issue 7](https://github.com/DCGenomics/NCBI_August_Hackathon_Push_Button_Genomics_Solution/issues/7)

Output:
- _VCF file with enhanced annotation_.  The output VCF files should include a "Predicted impact" column for each variant reports any reclassification of SnpEff's default impact values.  This would be an additional field in the INFO column.

This output will be used as input to the VCF filtering being implemented by @nicovbing in [Issue 8](https://github.com/DCGenomics/NCBI_August_Hackathon_Push_Button_Genomics_Solution/issues/8).

**Comment by eweitz** [comment]
@nicovbing developed a script to report "Predicted impact" in https://github.com/DCGenomics/NCBI_August_Hackathon_Push_Button_Genomics_Solution/commit/855d8ee58e775772650ecd9ece595fef9d39fe39.  Given that, I've reassigned the issue to him.

Let's close this issue once we've documented examples of how to call the script, and verified it works as expected.

**Comment by nicovbing** [comment]
HI Eric,

Sorry I have been busy (as everybody else).

Let me know when and how I can show how the script works.
Basically, it takes a vcf file with one line per variants per transcript in input and add the predicted impact (as: “;IMPACT=“) at the end of the line in the output.

Best,
NIco

**Comment by eweitz** [comment]
@nicovbing, can you provide an example command line call to your script using one of the VCF files in this repository, e.g. [/tests/test.GRCh38.chr22.vcf](https://github.com/DCGenomics/NCBI_August_Hackathon_Push_Button_Genomics_Solution/blob/master/tests/test.GRCh38.chr22.vcf)?  Thanks!

**Comment by nicovbing** [comment]
@eweitz 
The command is:
Rscript script_to_filter_VCF_impact_4.R YOUR_FILE_oneperline.vcf

The only problem is that my script takes VCF file with one variant per transcript per line. It does not support several transcripts per line.
I know there is a simple SNPeff function that modify the VCF file to get one per line but I don't know how it can be integrated in the whole pipeline.

Best,
Nico

**Comment by nicovbing** [comment]
@eweitz 
Sorry the command is:

 Rscript script_to_filter_VCF_impact.R YOUR_FILE_oneperline.vcf

**Comment by ghost** [comment]
Hi all - I have been busy trying to get a paper out the door since last Wednesday, sorry for disappearing.

The SNPeff flag referred to can be found here:

http://snpeff.sourceforge.net/SnpEff_manual.html#cmdline 
under database options  
-onlyTr            : Only use the transcripts in this file. Format: One transcript ID per line.

**Comment by CarlosBorroto** [comment]
Hi all, sorry I haven't contributed anything after the hackathon, I plan to do so.

@vlaufer this option doesn't seem to do what @nicovbing is asking. This option would limit the annotations to a list of transcripts provided on a **input** file with **one trasncript ID per line**. However, I think @nicovbing is asking for the **output** file to have only one transcript annotation per line.

One thing we could do is to follow @nicovbing recommendation to only annotate the canonical transcript, something I'm not convinced we should do, but it seems like @nicovbing has looked into this subject in more details than I.

`-canon                       : Only use canonical transcripts.`

**Comment by nicovbing** [comment]
@cjav yes I need a file in input with one transcript ID per line, which is the output of SNPeff.
It does not have to be only the canonical, my script will annotate all the missense variants. I just find it simpler if we get only the canonical.

**Comment by ghost** [comment]
@cjav - noted. In that case the simplest approach is to do the initial annotation that way. otherwise we add an extra step for no good reason.

Either tomorrow or Wednesday I will be able to finish the bash script I was working on to filter as well, maybe @nicovbing and I can compare scripts then.

**Comment by CarlosBorroto** [comment]
@nicovbing I see using just the canonical as a workaround to have a VCF output with only one transcript per line. I don't think there is an option that will duplicate every VCF line adding one annotation at a time.

**Comment by nicovbing** [comment]
@cjav There is an additional script in SNPeff that does that:
vcfEffOnePerLine.pl

From the documentation: "Splits EFF fields in a VCF file, creating mutiple lines, each one with only one effect.
Very useful for filtering with SnpSift."

**Comment by CarlosBorroto** [comment]
@nicovbing, I didn't know about that one. That could be an option.
